react/src/lib/components/SubsurfaceViewer/layers/triangle/privateTriangleLayer.ts

```diff
@@ -0,0 +1,206 @@
+import {
+    COORDINATE_SYSTEM,
+    Layer,
+    picking,
+    project,
+    phongLighting,
+    PickingInfo,
+    UpdateParameters,
+    LayerContext,
+} from "@deck.gl/core/typed";
+import {
+    createPropertyData,
+    LayerPickInfo,
+    PropertyDataType,
+} from "../utils/layerTools";
+import { Model, Geometry } from "@luma.gl/engine";
+import { DeckGLLayerContext } from "../../components/Map";
+import { ExtendedLayerProps } from "../utils/layerTools";
+import vsShader from "./vertex.glsl";
+import fsShader from "./fragment.fs.glsl";
+import vsLineShader from "./vertex_lines.glsl";
+import fsLineShader from "./fragment_lines.glsl";
+import GL from "@luma.gl/constants";
+
+export type GeometryTriangles = {
+    drawMode: number;
+    attributes: {
+        positions: { value: Float32Array; size: number };
+        TEXCOORD_0?: { value: Float32Array; size: number };
+        normals: { value: Float32Array; size: number };
+        vertex_indexs: { value: Int32Array; size: number };
+    };
+    vertexCount: number;
+    indices: { value: Uint32Array; size: number };
+};
+
+export type GeometryLines = {
+    drawMode: number;
+    attributes: {
+        positions: { value: Float32Array; size: number };
+    };
+    vertexCount: number;
+};
+
+export type Material =
+    | {
+          ambient: number;
+          diffuse: number;
+          shininess: number;
+          specularColor: [number, number, number];
+      }
+    | boolean;
+
+export interface privateTriangleLayerProps<D> extends ExtendedLayerProps<D> {
+    geometryTriangles: GeometryTriangles;
+    geometryLines: GeometryLines;
+    contours: [number, number];
+    gridLines: boolean;
+    color: [number, number, number];
+    smoothShading: boolean;
+    depthTest: boolean;
+}
+
+const defaultProps = {
+    data: ["dummy"],
+    contours: [-1, -1],
+    isContoursDepth: true,
+    gridLines: false,
+    color: [100, 100, 255],
+    coordinateSystem: COORDINATE_SYSTEM.CARTESIAN,
+    depthTest: true,
+};
+
+// This is a private layer used only by the composite TriangleLayer
+export default class privateTriangleLayer extends Layer<
```

Comment:
Would suggest sticking to variable case convention, so PascalCase for classes and camelCase for variables